src/engine/ruleset/integrations/suricata/decoders/suricata.yml

```
@@ -14,56 +14,229 @@ metadata:
-      - source.ip: $json_event.src_ip
-      - source.port: $json_event.src_port
-      - json_event.app_proto: downcase($json_event.app_proto)
+  # TODO: uncomment the lines below when the event field tag can be extracted
+  #- check: contains($tags, 'forwarded')
+  #  map:
+  #    - observer.product: Suricata
+  #    - observer.type: ids
+  #    - observer.vendor: OISF
 
-  - check: $is_good_appProto
+  - check: $suricata.app_proto == ftp-data
     map:
-      - network.protocol: $json_event.app_proto
+      - network.protocol: ftp
 
-  - check:
-      - json_event.app_proto: string_equal(ftp-data)
+  - check: >-
+        $suricata.app_proto != failed
+        AND $suricata.app_proto != template
+        AND $suricata.app_proto != template-rust
     map:
-      - network.protocol: ftp
-      - json_event.event_type: downcase($json_event.event_type)
-      - json_event.netwrok.transport: downcase($json_event.network.transport)
-      - json_event.proto: downcase($json_event.proto)
+      - network.protocol: $suricata.app_proto
 
-  - map:
-      - destination.bytes: $json_event.flow.bytes_toclient
-      - destination.mac: $json_event.ether.dest_mac
-      - destination.packets: $json_event.flow.pkts_toclient
-      - source.bytes: $json_event.flow.bytes_toserver
-      - source.mac: $json_event.ether.src_mac
-      - source.packets: $json_event.flow.pkts_toserver
-      - event.start: parse_date($json_event.flow.start, %FT%T%z)
-      - event.end: $json_event.flow.end
+  - check: >-
+        $suricata.event_type == http
+        AND $suricata.http.status < 400
+    map:
+      - event.outcome: success
 
-  - check:
-      - json_event.flow.state: string_equal(new)
+  - check: >-
+        $suricata.event_type == http
+        AND $suricata.http.status > 400
+    map:
+      - event.outcome: failure
+
+  - check: $suricata.flow.state == new
+    map:
+      - event.type: array_append(start)
+
+  - check: $suricata.flow.state == closed
     map:
       - event.type: array_append(end)
+
+  - check: is_number($suricata.alert.signature_id)
+    map:
+      - rule.id: to_string($suricata.alert.signature_id)
+
+  - check: $source.bytes >= 0 OR $destination.bytes >= 0
+    map:
+      - network.bytes: 0
+      - network.bytes: int_calculate(sum, $source.bytes, $destination.bytes)
+
+  - check: $source.packets >= 0 OR $destination.packets >= 0
+    map:
+      - network.packets: 0
+      - network.packets: int_calculate(sum, $source.packets, $destination.packets)
+
+  - check:
+      - suricata.alert.action: is_not_null()
+    map:
+      - event.type: array_append($suricata.alert.action)
+
+  - check: $network.protocol == http
+    map:
+      - url.domain: $destination.domain
+
+  # DNS
+  - check: $network.protocol == dns
+    map:
+      - dns.id: to_string($suricata.dns.id)
+      - dns.response_code: $suricata.dns.rcode
+      - dns.type: $suricata.dns.type
+      - wazuh.dns.question.original: $suricata.dns.rrname
```

Change field `wazuh.dns.question.original` to `suricata.dns.question`
```suggestion
      - suricata.dns.question.original: $suricata.dns.rrname
```